tincr/io/device/family_info.tcl

```diff
@@ -0,0 +1,773 @@
+            }
+        }
+        
+        # for each site, go through and try to determine more compatible sites 
+        dict for {type site} $site_type_map {
+                
+            if { [llength [get_bels -of $site -quiet]] == 1 && [dict exists $sitetype_to_cell_map $type] } {
+                set cell_list [dict get $sitetype_to_cell_map $type]
+                
+                # create a list of candidate sites that could possibly be compatible
+                set candidate_sites [list]
+                set i 0
+                foreach cell $cell_list {
+                    if {$i ==0} {
+                        set candidate_sites [dict get $cell_to_sitetype_map $cell]
+                        incr i
+                    } else {
+                        set candidate_sites [::struct::set intersect $candidate_sites [dict get $cell_to_sitetype_map $cell]]
+                    }
+                }
+                
+                # check if the candidates are valid compatible sites
+                foreach candidate_type $candidate_sites {
+                    
+                    set candidate_cell_count  [llength [dict get $sitetype_to_cell_map $candidate_type]]
+                    
+                    # check to see if the candidate is a valid compatible site
+                    if { $candidate_type != $type && $candidate_cell_count >= [llength $cell_list] } {
+                        
+                        # get the list of currently compatible site
+                        set current_list [list]
+                        if {[dict exists $compatible_type_map $type]} {
+                            set current_list [dict get $compatible_type_map $type]
+                        }
+                        
+                        # If the candidate is compatible, then add it to the compatible map
+                        if {[lsearch $current_list $candidate_type] == -1} {
+                            dict lappend compatible_type_map $type $candidate_type
+                        }
+                        
+                        # For alternate-only compatible types, mark the default type as compatible as well.
+                        if {[::struct::set contains $alternate_site_set $candidate_type] } {
+                            set default_site [dict get $site_type_map $candidate_type]
+                            reset_property MANUAL_ROUTING $default_site
+                            set default_type [get_property SITE_TYPE $default_site]
+                            
+                            if {[lsearch $current_list $default_type] == -1} {
+                                dict lappend compatible_type_map $type $default_type
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    
+    return $compatible_type_map
+}
+
+## Parses the VSRT "addedBels.txt" file for the current architecture,
```

Comment:
I thought VSRT output primitive definition files? What it this `addedBels.txt` file?